tezos/context/src/working_tree/working_tree.rs

```diff
@@ -1012,6 +1099,163 @@ impl WorkingTree {
         hash_directory(root, repository, &storage, &strings).map_err(MerkleError::from)
     }
 
+    fn traverse_working_tree_recursive(
```

Comment:
@sebastiencs please add a comment here mentioning that this is only used by the tool, or move the implementation to a separate file so that it is obvious. That way it will be clear that the unwraps and panics here are not much of an issue.